liblangutil/Scanner.cpp

```diff
@@ -826,31 +852,50 @@ Token Scanner::scanString(bool const _isUnicode)
 	char const quote = m_char;
 	advance();  // consume quote
 	LiteralScope literal(this, LITERAL_TYPE_STRING);
-	while (m_char != quote && !isSourcePastEndOfInput() && !isUnicodeLinebreak())
+	// for source location comments we allow multiline string literals
+	while (m_char != quote && !isSourcePastEndOfInput() && (!isUnicodeLinebreak() || m_kind == ScannerKind::SpecialComment))
 	{
 		char c = m_char;
 		advance();
-		if (c == '\\')
+
+		if (m_kind == ScannerKind::SpecialComment)
 		{
-			if (isSourcePastEndOfInput() || !scanEscape())
-				return setError(ScannerError::IllegalEscapeSequence);
+			if (c == '\\')
+			{
+				if (isSourcePastEndOfInput())
+					return setError(ScannerError::IllegalEscapeSequence);
+				bool const validEscape = scanEscape(false /* _rejectInvalidEscapes */);
```

Comment:
So just that I understand correctly: the only thing you need to do here is to check for escaped quotes, right? Couldn't you do that here directly instead of the entire refactoring of ``scanEscape``? We don't actually do anything with the source quote in the snippets other than discarding them, so not sure why we'd go out of our way to deal with all sorts of escape sequences here...

Reply:
The change was based off of [this comment](https://github.com/ethereum/solidity/pull/15209#discussion_r1657457198). I am completely fine with just checking for escaped quotes as well - all depends on the intended use of the scanner mode beyond this pr I guess.

Reply:
``scanEscape(false)`` is generally rather weird, though, isn't it? Either you *want* to properly handle all kinds of escape sequences are you don't, but handling *some* properly and ignoring others is strange, isn't it :-)?
I think the simpler solution to just *only* handle escaped quotes here matches more directly what we need now, and keeps the PR simpler - and if we ever want to use this for anything else we can revisit this then...

Reply:
*Was* a code snippet runinng to EOF here actually even an error before the change :-)?

Reply:
It would have reported an error because of missing end quote (`"literal\` - here the `\` is no longer part of the match).

Reply:
Yeah, the previous condition for error 1544 is rather convoluted, but looks like that's what it did.